### PR TITLE
Lpd 19827 - Add support to ERC to Commerce Order delivery API

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-19827

### CODE REVIEW

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -729,6 +823,60 @@ paths:
                     description:
                         Unexpected error
             tags: ["PlacedOrder"]
+    /placed-order-comments/by-external-reference-code/{externalReferenceCode}:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:826]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -788,6 +936,110 @@ paths:
                         Unexpected error
             summary: Gets a Note by ID.
             tags: ["PlacedOrderComment"]
+    /placed-order-items/by-external-reference-code/{externalReferenceCode}:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:939]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 3]
```diff
@@ -788,6 +936,110 @@ paths:
...
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            tags: ["PlacedOrderItem"]
+    /placed-order-items/by-external-reference-code/{externalReferenceCode}/placed-order-item-shipments:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:989]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 4]
```diff
@@ -902,6 +1154,332 @@ paths:
                     description:
                         Unexpected error
             tags: ["PlacedOrderItemShipment"]
+    /placed-orders/by-external-reference-code/{externalReferenceCode}:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:1157]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 5]
```diff
@@ -902,6 +1154,332 @@ paths:
...
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            tags: ["PlacedOrder"]
+    /placed-orders/by-external-reference-code/{externalReferenceCode}/payment-url:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:1208]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 6]
```diff
@@ -902,6 +1154,332 @@ paths:
...
+                  schema:
+                      type: string
+            responses:
+                200:
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+            tags: ["PlacedOrder"]
+    /placed-orders/by-external-reference-code/{externalReferenceCode}/placed-order-billing-address:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:1229]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 7]
```diff
@@ -902,6 +1154,332 @@ paths:
...
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            tags: ["PlacedOrderAddress"]
+    /placed-orders/by-external-reference-code/{externalReferenceCode}/placed-order-comments:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:1280]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 8]
```diff
@@ -902,6 +1154,332 @@ paths:
...
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            summary: Gets a list of Comments related to an Order.
+            tags: ["PlacedOrderComment"]
+    /placed-orders/by-external-reference-code/{externalReferenceCode}/placed-order-items:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:1349]: by-externalReferenceCode (not by-external-reference-code)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 9]
```diff
@@ -902,6 +1154,332 @@ paths:
...
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            tags: ["PlacedOrderItem"]
+    /placed-orders/by-external-reference-code/{externalReferenceCode}/placed-order-shipping-address:
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:1432]: by-externalReferenceCode (not by-external-reference-code)

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by smottal** [comment]
Hey @matyaswollner , some changes needed. They should be quickly because don't impact too much on the code itself but mostly on the autogenerated classes

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:report:2560827776

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 10]
```diff
@@ -788,6 +941,110 @@ paths:
                         Unexpected error
             summary: Gets a Note by ID.
             tags: ["PlacedOrderComment"]
+    /placed-order-items/by-externalReferenceCode/{externalReferenceCode}:
+        get:
+            description:
+                Retrieve information of the given Placed Order.
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:947]: order item

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml [hunk 11]
```diff
@@ -788,6 +941,110 @@ paths:
...
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Error"
+                    description:
+                        Unexpected error
+            tags: ["PlacedOrderItem"]
+    /placed-order-items/by-externalReferenceCode/{externalReferenceCode}/placed-order-item-shipments:
+        get:
+            description:
+                Retrieve shipments of the given Placed Order Item.
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/rest-openapi.yaml:997]: not sure if placed order item should be capitalized

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/src/main/java/com/liferay/headless/commerce/delivery/order/internal/resource/v1_0/PlacedOrderResourceImpl.java [hunk 12]
```diff
@@ -86,6 +124,47 @@ public PlacedOrder getPlacedOrder(Long placedOrderId) throws Exception {
...
+			throw new NoSuchOrderException(
+				"Unable to find order with external reference code " +
+					externalReferenceCode);
+		}
+
+		if (commerceOrder.isOpen()) {
+			throw new NoSuchOrderException();
+		}
+
+		return _toPlacedOrder(commerceOrder.getCommerceOrderId());
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/src/main/java/com/liferay/headless/commerce/delivery/order/internal/resource/v1_0/PlacedOrderResourceImpl.java:146]: why here is not called the get method with id?

**Review by andrea-ale-sbarra — APPROVED** [review]
overall looks good, just few notes before forwarding.
Thank you!

**Comment by matyaswollner** [comment]
@andrea-ale-sbarra I've pushed the suggested changes. Thank you for the review!

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**